src/main/scala/pl/project13/scala/akka/raft/protocol/StateMetadata.scala

```diff
@@ -74,9 +79,9 @@ private[protocol] trait StateMetadata extends Serializable {
 
     def withVoteFor(term: Term, candidate: ActorRef) = copy(votes = votes + (term -> candidate))
 
-    def forLeader: LeaderMeta        = LeaderMeta(clusterSelf, currentTerm, config)
-    def forFollower: Meta            = Meta(clusterSelf, currentTerm, config, Map.empty)
-    def forNewElection: ElectionMeta = this.forFollower.forNewElection
+    def forLeader: LeaderMeta = LeaderMeta(clusterSelf, currentTerm, config)
+    def forFollower(term: Term = currentTerm): Meta = Meta(clusterSelf, term, config, Map.empty)
```

Comment:
Nice catch! Also though about that. There is a trade off between efficiency and readability and I just wanted to remove unnecessary copying of objects if its possible. Surely, I know about premature optimizations :) I think you're right there is also one more place where same "side effect" is hidden.